dali/c_api_2/error_handling.cc

```diff
@@ -104,10 +104,10 @@ daliResult_t HandleError(std::exception_ptr ex) {
     return SetLastError(DALI_ERROR_OUT_OF_MEMORY, e.what());
   } catch (std::bad_alloc &e) {
     return SetLastError(DALI_ERROR_OUT_OF_MEMORY, e.what());
-  } catch (dali::invalid_key &e) {
-    return SetLastError(DALI_ERROR_INVALID_KEY, e.what());
   } catch (std::out_of_range &e) {
     return SetLastError(DALI_ERROR_OUT_OF_RANGE, e.what());
+  } catch (std::invalid_argument &e) {
+    return SetLastError(DALI_ERROR_INVALID_ARGUMENT, e.what());
```

Comment:
Is there a particular reason for reordering the cases here?

Reply:
The order does matter. I think it was formally correct before (I reordered those during debugging), but while at it, I realized that the ordering wasn't particularly clear, so I kept the new one.